src/config.ts

```diff
@@ -296,43 +296,44 @@ export class MirrordConfigManager {
     if (this.active) {
       // User has selected a config (via active config button).
       new NotificationBuilder()
-        .withMessage("using active mirrord configuration")
+        .withMessage("Using active mirrord configuration.")
         .withOpenFileAction(this.active)
         .withDisableAction("promptUsingActiveConfig")
         .info();
+
       return this.active;
     } else if (config.env?.["MIRRORD_CONFIG_FILE"]) {
       // Get the config path from the env var.
-      return vscode.Uri.parse(`file://${config.env?.["MIRRORD_CONFIG_FILE"]}`, true);
+      const configFromEnv = vscode.Uri.parse(`file://${config.env?.["MIRRORD_CONFIG_FILE"]}`, true);
+
+      new NotificationBuilder()
+        .withMessage(`Using mirrord configuration from env var "MIRRORD_CONFIG_FILE".`)
+        .withOpenFileAction(configFromEnv)
+        .withDisableAction("promptUsingActiveConfig")
+        .info();
+
+      return configFromEnv;
+
     } else if (folder) {
-      let predefinedConfig = await MirrordConfigManager.getDefaultConfig(folder);
-      if (predefinedConfig) {
+      const configFromMirrordFolder = await MirrordConfigManager.getDefaultConfig(folder);
+
+      if (configFromMirrordFolder) {
         new NotificationBuilder()
-          .withMessage("using a default mirrord config")
-          .withOpenFileAction(predefinedConfig)
+          .withMessage(`Using mirrord configuration from ".mirrord" folder.`)
+          .withOpenFileAction(configFromMirrordFolder)
           .withDisableAction("promptUsingDefaultConfig")
-          .warning();
-        return predefinedConfig;
+          .info();
+
+        return configFromMirrordFolder;
       } else {
         return null;
       }
     } else {
-      folder = vscode.workspace.workspaceFolders?.[0];
```

Comment:
So the case where `folder` is `undefined` but `vscode.workspace.workspaceFolders?.[0]` is defined doesn't really exist?

Reply:
Well, I'm not even sure how to trigger this.

Maybe if you open a file without a `.vscode` folder nearby? But at that point, you probably don't have a `.mirrord` anyways?

Reply:
Which is probably why we throw an error in that case, no?

Reply:
hmm, let me try to trigger this and see what happens, brb.

Reply:
Alright got it to trigger, this seems more like a UX question now.

If you open a file, instead of a folder, there is no .vscode, and when you launch the app with mirrord, vscode asks you the type of app to launch, then it starts.

For the mirrord part:

If we just return null here, mirrord works, but if we return an exception, it doesn't (as expectec). Do we want to force the user to solve this, or just let it work with a default config?

Reply:
I see some value in just letting it work with a default config. If you just want to try mirrord out in some random file, or maybe to run some quick script-ish thing in the cluster, you can just create a `quick-run.whatever`, then run it with mirrord.

Reply:
Added notification that we're using a default config.

Did this as a separate function, so we can return the same message whenever we would return `null`, wdyt?

Reply:
Eyal said we should not show a warning on when running normally without a config file. The case Aviram said should have a notification is when vscode is open on a single file instead of a library. In that case we should let the user know that since they are working on a single file mirrord cannot use a configuration file, even if one exists.

Reply:
And to the original question: Does the case where `folder` is `undefined` but `vscode.workspace.workspaceFolders?.[0]` is defined not exist?

Reply:
I'll check it, but I don't think so, unless vscode creates a hidden `workspaceFolders`, which doesn't make sense.

> not show a notification when running normally without a config file

Oh yeah, will remove this notification from there.

Reply:
> folder is undefined but vscode.workspace.workspaceFolders?.[0] is defined not exist?

Yeah, I think this case **doesn't** exist, or at least I'm not sure how to build this scenario? Maybe if you have some very specific config (I'm just imagining stuff here) like a global vscode config that sets workspace to some path? This seems very far from normal usage though. Don't think we care about it.

Reply:
Updated, now we only `info` when running in a single file.